### ISSUE TITLE
[Panic] ruff fix-import

### ISSUE THREAD

**Issue by justinchuby** [charliermarsh/ruff#3033]
### File

The file is `https://github.com/onnx/onnx/blob/7f0a9f2537348e7e53a825e0b716db1dbc997e47/onnx/backend/test/case/node/where.py`

### pyproject.toml

```toml
[tool.ruff]
target-version = "py37"
select = [
    "E", # pycodestyle
    "F", # Pyflakes
    "W", # pycodestyle
    "B", # flake8-bugbear
    "N", # pep8-naming
    "YTT", # flake8-2020
    "RUF", # Ruff-specific rules
    "TID252", # Relative imports
]
ignore = [
    "B9", # Opinionated bugbear rules
    "E402", # warns on imports coming after statements. There are important use cases
            # like demandimport (https://fburl.com/demandimport) that require statements
    "E501",
    "N803", # Argument casing
    "N806", # Variable name in function should be lowercase
]
ignore-init-module-imports = true
unfixable = [
    "F401", # Unused imports
]

[tool.ruff.flake8-tidy-imports]
# Disallow all relative imports.
ban-relative-imports = "all"
```

### Command ran

```sh
RUST_BACKTRACE=full ruff --fix-only --exit-zero --config=pyproject.toml /Users/justinc/Documents/GitHub/onnx/onnx/backend/test/case/node/where.py
```

Error:

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/ruff/src/rules/flake8_tidy_imports/relative_imports.rs:102:54
stack backtrace:
   0:        0x1054e8ed8 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2990a6f24ccff6db
   1:        0x104edb668 - core::fmt::write::h91497fd291c8b104
   2:        0x1054e3ea0 - std::io::stdio::_eprint::hfdc1bd7888eaa0e7
   3:        0x1054eca70 - std::panicking::default_hook::hdfe992d5fb29a991
   4:        0x1054ec798 - std::panicking::default_hook::hdfe992d5fb29a991
   5:        0x1052796d4 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
   6:        0x1054ed02c - std::panicking::rust_panic_with_hook::h2eb0e4b718773521
   7:        0x1054eceb8 - <std::panicking::begin_panic_handler::StrPanicPayload as core::panic::BoxMeUp>::get::h3f7cfefea3e19c93
   8:        0x1054eb8c4 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2990a6f24ccff6db
   9:        0x1054ecc74 - _rust_begin_unwind
  10:        0x1055b0eec - core::panicking::panic_fmt::ha46aa9af97eb193d
  11:        0x1055b0e0c - core::panicking::panic::h5713e7735cef5fa8
  12:        0x10531cce4 - <ruff::checkers::ast::Checker as ruff::ast::visitor::Visitor>::visit_stmt::h4b593c8c33a76cb3
  13:        0x105333440 - <ruff::checkers::ast::Checker as ruff::ast::visitor::Visitor>::visit_body::h0fc88a428b1d49d5
  14:        0x10534a4d8 - <ruff::checkers::ast::Checker as ruff::ast::visitor::Visitor>::visit_body::h0fc88a428b1d49d5
  15:        0x10535a298 - ruff::linter::check_path::h842469eb58db80b1
  16:        0x105364d20 - ruff::linter::lint_fix::h929bb0874a600095
  17:        0x105266a30 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  18:        0x10520086c - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  19:        0x105264aa4 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  20:        0x10527d370 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  21:        0x105274640 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  22:        0x10527eb30 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  23:        0x1051d47d8 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  24:        0x1051d4c80 - <reqwest::redirect::TooManyRedirects as core::fmt::Debug>::fmt::h082b790308395cc4
  25:        0x1054debc8 - std::rt::lang_start_internal::h94fa211da9d06d56
  26:        0x10528c58c - _main
fish: Job 1, 'RUST_BACKTRACE=full ruff --fix-…' terminated by signal SIGABRT (Abort)
```

### Version

ruff 0.0.247

**Comment by charliermarsh** [comment]
Taking a look...

**Comment by charliermarsh** [comment]
Just confirmed locally that this was fixed by #2990. Will cut a new release today.

**Comment by justinchuby** [comment]
Thanks!

**Comment by charliermarsh** [comment]
Thanks for filing!
